### ISSUE TITLE
Regression in several platform_views_scroll_perf_ios__timeline_summary benchmarks

### ISSUE THREAD

**Issue by kf6gpe** [flutter/flutter#53266]
Looks like it began with this engine roll:
https://github.com/flutter/flutter/pull/53200

99th_percentile_frame_rasterizer_time_millis, average_frame_rasterizer_time_millis, and worst_frame_rasterizer_time_millis are affected.

![Screen Shot 2020-03-25 at 8 42 31 AM](https://user-images.githubusercontent.com/1435716/77555821-f0985c00-6e74-11ea-9b2a-37a814bdc41a.png)

**Comment by kf6gpe** [comment]
This apparently was somewhat expected. cc @blasten to work on reasons & rebaseline as needed. Thanks.

**Comment by blasten** [comment]
This is due to resizing surfaces in each frame on iOS. This is very expensive and likely requires a workaround. cc @chinmaygarde

**Comment by blasten** [comment]
I will remove the today label for now, and close the issue once the perf issue is resolved or the benchmark updated.

**Comment by dnfield** [comment]
@blasten - would we really be ok going to stable with this regression?  Won't this affect all iOS users for webview, maps, etc?  This seems like a pretty significant regression.

**Comment by dnfield** [comment]
At the very least, we should be documenting here why we think this is ok and why it's worth the significant drop in the performance.

**Comment by blasten** [comment]
The benchmark is measuring a worst case situation. I will bring this up to stakeholders to figure out whether it should block or not in case we don’t find a way to work around it.  I had an idea about resizing the overlays without changing the surface size.

**Comment by chinmaygarde** [comment]
@blasten Till the setter of the frame overlay still has the issue where minor inconsistencies due to rounding cause frequent resizes, this worst case behavior will be seen in all cases. Let's address that regardless of whether this is reflected in this benchmark (where the platform view is in the scroll view). It may be a good idea to author a benchmark for a platform view that does not sit in a scroll view to specifically check for unnecessary updates to the overlay frame.

> This is very expensive and likely requires a workaround.

I don't know of any workaround that will ameliorate this in the near future. For now, this is the price of the feature that unblocks the use-case. There should a requisite increase in device memory use as well for the additional buffered backing stores per layer.

> At the very least, we should be documenting here why we think this is ok and why it's worth the significant drop in the performance.

The document is internal, I'll ping the authors internally to see if there is a public version of the doc.

**Comment by blasten** [comment]
@chinmaygarde I was thinking of using a wrapper UIView that masks the overlay with the fixed size. Do you think that won't work?

**Comment by chinmaygarde** [comment]
Assuming the backing stores of the masked content have to be fullscreen. It will work in that this benchmark will get better but the memory situation will be quite a bit worse.

**Comment by blasten** [comment]
@chinmaygarde before my changes, each platform view always had a UIView and backing surface.  Now, it only creates a UIView overlay if there's Flutter UI on top of the platform view. There's a max of two UIViews, so if they are more overlays, they are combined into a single one (just like before my changes).

**Comment by chinmaygarde** [comment]
Say there is a platform view with Flutter rendered contents on top (by UIKit z-index) of it but not obstructing it. These contents are above and below (screen space) the platform view itself. Now that we want the platform view to be unobstructed, there will need to be two overlays for the contents on top and bottom. If you want to minimize the resizing of those overlays due by using UIView masks, those overlays now need to be full sized. That was the memory use I was concerned about. Did I misunderstand [your comment](https://github.com/flutter/flutter/issues/53266#issuecomment-604041704)?

**Comment by blasten** [comment]
> Say there is a platform view with Flutter rendered contents on top (by UIKit z-index) of it but not obstructing it. 

This should result in zero overlays since the Rtree's search for intersecting rects returns empty.

**Comment by blasten** [comment]
cc @amirh @cyanglaz

**Comment by chinmaygarde** [comment]
> This should result in zero overlays since the Rtree's search for intersecting rects returns empty.

Ah. I missed that. Thanks for clearing that up. Yeah, then [your approach](https://github.com/flutter/flutter/issues/53266#issuecomment-604041704) sounds promising.

**Comment by chinmaygarde** [comment]
Given that we have a plan that does not require a rebaseline, can we temporarily back out the change so that benchmarks are back to normal and close this bug? The patch after accounting for the resizing can then land in one shot.

**Comment by dnfield** [comment]
I think as this is today, it'll be a surprising regression for users who are happy on stable.

**Comment by dnfield** [comment]
Reapplying today to block the google3 roll until we have agreement on that - if there's something I'm missing and we really have to move forward with this we can remove it.

**Comment by blasten** [comment]
I don't think we can revert at the moment. We have folks doing other type of testing.  I will have a patch shortly to address this issue.

**Comment by dnfield** [comment]
If we have a fast fix forward that's probably fine too.  Just should make sure we coordinate with roller(s) to not tag revisions missing the fix for dev.

**Comment by blasten** [comment]
cc @pcsosinski

**Comment by chinmaygarde** [comment]
@blasten Let's stick to the documented guidance on the "Handling TODAY Issues" section in go/flutter-engine-sheriff which is essentially re-baseline or revert. As Dan mentioned, this has repercussions for users and roller. I'd also rather have the patch and subsequent fixes in one single commit in case we run into further issues.

**Comment by pcsosinski** [comment]
agree we should rollback and then re-commit with the patch.

**Comment by blasten** [comment]
I added this benchmark last week to monitor the performance of some code paths. I created an intentionally difficult case that may not reflect a real case. Android for example shows that the cost of these paths is almost 2x of the iOS ones even after my patch. 

This is the judgment I was following for removing the "today" label.  PR reviewer time is very limited. I didn't see a section on the doc about that (http://go/review-speed), so I can provide better estimates about deliverables.
